fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/insert/InsertUtils.java

```diff
@@ -395,16 +411,21 @@ private static Expression castValue(Expression value, DataType targetType) {
      */
     public static TableIf getTargetTable(Plan plan, ConnectContext ctx) {
         UnboundLogicalSink<? extends Plan> unboundTableSink;
-        if (plan instanceof UnboundTableSink) {
-            unboundTableSink = (UnboundTableSink<? extends Plan>) plan;
-        } else if (plan instanceof UnboundHiveTableSink) {
-            unboundTableSink = (UnboundHiveTableSink<? extends Plan>) plan;
-        } else if (plan instanceof UnboundIcebergTableSink) {
-            unboundTableSink = (UnboundIcebergTableSink<? extends Plan>) plan;
+        Plan sinkPlan = plan;
+        if (plan instanceof LogicalStatementHint) {
```

Comment:
too many `if (plan instanceOf LogicalStatementHint) {...} else {...}` in different place. Maybe it is better to save hint in InserIntoTableCommand, and add it back to plan tree before do real plan in `initPlan`. or do `eliminate logical statement` at very begining in InsertIntoTableCommand